### ISSUE TITLE
Need ability to display SSH stdout in some situations (e.g. tail logs)

### ISSUE THREAD

**Issue by carlesso** [mattbrictson/airbrussh#3]
Hi, thanks for the great work, airbrussh really makes capistrano looks better! :)

I've a question. I've a simple task like this:

```
namespace :logs do
  desc 'Tailf production logs'
  task :tail do
    on roles(:app) do
      set :format, :pretty
      execute :tail, '-f', release_path.join('log', "#{fetch(:rails_env, 'production')}.log")
    end
  end
end
```

But now with airbrussh all the logs goes to the `log/capistrano.log`. Is there a way to change this?

**Comment by mattbrictson** [comment]
Good question. Airbrussh currently mutes all stdout and stderr output for brevity. The `set :format, :pretty` statement doesn't work within your `logs:tail` task definition because Capistrano configures the logger at startup time and after that it is locked.

Could you do something like this in your `deploy.rb`?

```ruby
set(:format, :pretty) if ENV["CAP_PRETTY"]
```

Then when you need to tail the production logs you could specify that env variable like this:

```
CAP_PRETTY=1 cap production logs:tail
```

An advantage of this approach is that you don't have to modify your `logs:tail` task.

**Comment by carlesso** [comment]
This way works fine, but I was trying to make a gem with some couple of tasks in it, without the need to change the `deploy.rb` in the project.
I'll inspect your formatter further to look for a solution, meanwhile if you have any hint, more then welcome!

**Comment by mattbrictson** [comment]
In `Airbrussh::Formatter#write_command`, you could check whether the command has stdout data and write that data to the underlying output. This behavior could be gated, perhaps by a setting that you change inside your tasks.

Here's how the default `:pretty` formatter does it:

https://github.com/capistrano/sshkit/blob/v1.7.1/lib/sshkit/formatters/pretty.rb#L31

The hard part is implementing the gate without tightly coupling your gem to Airbrussh. Maybe your gem could do something like:

```ruby
# Tell Airbrussh to begin passing through stdout so we can see it on the console.
# Airbrussh may be an old version or might not be installed, so test for that.
Airbrussh.unmute_stdout if defined?(Airbrussh) && Airbrussh.respond_to?(:unmute_stdout)
```

**Comment by mattbrictson** [comment]
Alternatively, we could change Airbrussh to check the existing `:log_level` setting. If set to `:debug`, then show command output; if `:info`, use the current behavior. (Although, again, I don't think Capistrano lets you change `:log_level` after startup.)

There are a bunch of different avenues we could take this. I'm open to suggestions!

**Comment by justindowning** [comment]
For my uses, could we add a flag that could be called per task to not output the `write_banner`? This would encourage more machine readable output for tasks the depend on it. Thanks! https://github.com/mattbrictson/airbrussh/blob/e3a8c5aeaaf8345ade13ddc981fb5b924f9cfb37/lib/airbrussh/formatter.rb#L55-L59

**Comment by mattbrictson** [comment]
@justindowning I am going to move your suggestion into a separate issue.

**Comment by mattbrictson** [comment]
@carlesso what do you think of a hypothetical implementation that uses configuration options like these? This feels more self-explanatory than `log_level = :debug`.

```ruby
# Default. Output of SSH commands is not displayed.
config.command_output = false

# Display stdout of SSH commands. Stderr is not displayed.
config.command_output = :stdout

# Display stderr of SSH commands. Stdout is not displayed.
config.command_output = :stderr

# Display all SSH command output.
config.command_output = [:stdout, :stderr]
# or
config.command_output = true
```

**Comment by carlesso** [comment]
I like that. Really self explanatory.

**Comment by mattbrictson** [comment]
@carlesso OK I will work on implementing that.

**Comment by mattbrictson** [comment]
Implemented in 42a16ff.
